framework/src/timeintegrators/NewmarkBeta.C

```diff
@@ -0,0 +1,80 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NewmarkBeta.h"
+#include "NonlinearSystem.h"
+
+registerMooseObject("MooseApp", NewmarkBeta);
+
+template <>
+InputParameters
+validParams<NewmarkBeta>()
+{
+  InputParameters params = validParams<TimeIntegrator>();
+  params.addClassDescription(
+      "Computes the first and second time derivative of variable using Newmark-Beta method.");
+  params.addParam<Real>("beta", 0.25, "beta value");
+  params.addParam<Real>("gamma", 0.5, "gamma value");
+  return params;
+}
+
+NewmarkBeta::NewmarkBeta(const InputParameters & parameters)
+  : TimeIntegrator(parameters),
+    _beta(getParam<Real>("beta")),
+    _gamma(getParam<Real>("gamma")),
+    _u_dot_old(*_sys.solutionUDotOld()),
+    _u_dotdot(_sys.solutionUDotDot()),
+    _u_dotdot_old(*_sys.solutionUDotDotOld()),
+    _du_dotdot_du(_sys.duDotDotDu())
+{
+}
```

Comment:
As in the lecture notes, there are some restrictions on beta, gamma required to guarantee stability and accuracy, namely: `2*beta >= gamma >= 1/2` for stability and second order only for `gamma=1/2`. Do you think we should add some checks for these so that users don't accidentally use invalid values?

Reply:
I will add the condition on gamma for stability. However, Gamma = 1/2 is not really required. When using HHT-alpha method, we set gamma higher than 1/2 and it is still second order accurate (page 6 of lecture notes).

Reply:
For HHT you may be right, but for Newmark-beta the last sentence of paragraph 1 on pg. 5 of the lecture notes says: "For gamma=1/2 the Newmark-beta method is at least second-order accurate; it is first order accurate for *all* other values of gamma."